### PR TITLE
Weather login customizations module for limiting access to basic auth features

### LINKED ISSUE #1006: Remove basic auth from the site and decide what we want out of a Drupal user/login page
### Description

Followup to #952. We created a new login page for redirecting users upon logout (or pointing them to it as our login page) as part of this ticket, but as of now it is undesigned/styled. We can accomplish a lot by just removing entirely the root login username/password and that helps with simplifying design. However, we also need to figure out what we want from an interstitial page at all. 

### Questions and Considerations

Currently looks like: 
Screenshot 2024-04-03 at 10 06 52 AM

### Acceptance Criteria
- [x] Remove basic auth from form 
- [x] Remove /register and /password routes
- [ ] "Secure" the root account https://www.drupal.org/docs/administering-a-drupal-site/security-in-drupal/securing-the-admin-super-user-1
- [x] Document process of generating a login for the root user for our system admins
- [x] Ensure everyone on the team has appropriate admin permissions in every environment so no need for use of root account

### PR DESCRIPTION
**Pull request by loganmcdonald-noaa**
## What does this PR do? 🛠️
Fixes #1006. We are adding a small module for the login experience that allows us to reroute a couple things and constrain login to SSO. In local dev and design test environments, we will still use username/password but in all other environments, SSO only. This also removes the /user/password and /user/register endpoints from public consumption. 

Major shoutout to USA.gov for their open source work that helped me figure out this module!

## For reviewers
ATTENTION! @eric-gade @greg-does-weather  @igorkorenfeld this requires a change to local settings! Make sure to add the changed line or recopy settings.dev.yml! 

## Screenshots (if appropriate): 📸
Environments with basic auth see: 
Screenshot 2024-05-06 at 4 08 32 PM

Environments with SAML enabled see: 
Screenshot 2024-05-06 at 4 08 24 PM

### CODE REVIEW

File: web/sites/example.settings.dev.php [hunk 1]
```diff
@@ -845,3 +845,5 @@
 ];
 $settings["config_sync_directory"] = "/opt/drupal/web/config/sync/";
 $config["config_split.config_split.local"]["status"] = true;
+$settings["weather_login_local_form"] = true;
```
Comment (loganmcdonald-noaa) [inline web/sites/example.settings.dev.php:848]: This is the setting you have to update!!!

File: Makefile [hunk 2]
```diff
@@ -80,6 +80,11 @@ zap-containers:
 	docker compose stop
 	docker compose rm -f
 
+scorched-earth:
+	docker stop $$(docker ps -a -q)
+	docker rm $$(docker ps -a -q)
+	docker system prune -f
+
```
Comment (greg-does-weather) [inline Makefile:87]: 😂

File: web/modules/weather_login/weather_login.module [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    $loginPath = $config->get('sso_login_path');
+    $loginText = $config->get('sso_login_text');
+    $cancelPath = $config->get('sso_cancel_path');
+    $cancelText = $config->get('sso_cancel_text');
+
+    // Allow override for local login forms via $settings[] in settings.*.php
+    $settings_state = Settings::get('weather_login_local_form', 0);
+    if ($settings_state || $state) {
+        $state = 1;
+    }
```
Comment (greg-does-weather) [inline web/modules/weather_login/weather_login.module:26]: `$state` isn't defined prior, so I'm curious if the `or` is necessary.

**Review by greg-does-weather — APPROVED** [review]
Wow. This is cool! There's a lot of stuff going on in here. Nicely done!

**Pull request merged**